experimental/dds/tree2/src/feature-libraries/editable-tree-2/README.md

```diff
@@ -0,0 +1,84 @@
+Note that `enumerable` has nothing to do with iterable: Iterable is not relevant to this section.
+
+Despite this tree API primarily being a TypeScript API, and TypeScript types having no way to indicate if members are `enumerable` or `own`, these details matter, even to TypeScript users.
+
+For example, TypeScript assumes all members of interfaces are `enumerable` and `own` when using the [Object spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) operator.
+Typescript also assumes all non-method members of class are `enumerable` and `own`, but allows assigning the class to an interface where it will start treating the methods as `enumerable` and `own` despite it being the same object.
+This compile time behavior often does not match what happens at runtime, so users of TypeScript are likely to run into issues despite their code compiling just fine if they rely on Object spread, or make assumptions about what properties are `enumerable` or `own`.
+
+Many existing libraries users of tree are likely to work with will also make assumptions about `enumerable` or `own`.
+
+So in addition to the TypeScript types, a separate decision needs to be made about what guarantees this library will make about `enumerable` and `own` properties.
+
+This library guarantees then when traversing from a root `UntypedEntity` via `enumerable` `own` properties:
+
+-   All callable members are `inherited` (not `own`) or not `enumerable` and thus work like class methods. Note that TypeScript's type checking will get this wrong due to the API using interfaces.
+-   When starting at a root node or fields, there ie exactly one way to traverse to (or past in some unboxed cases) every node and field under it via only enumerable own properties.
+-   Every leaf node's value within the tree will be reachable, either from its node, it as its node (in the unboxed case). Note that values are assumed to be immutable, and if multiple leaves hold structurally identical objects as values they may or may not be shared and this difference is not considered significant. TODO: determine how node's assert.deepEqual compares these cases.
+-   Every node traversed has an unambiguous type, either implied by its position and parent's schema (for unboxed cases) and/or from an `enumerable` `own` property containing the schema's identifier.
+-   No cycles will be encountered, with the exception of any `FluidHandle` stored as part of serializable values on `LeafNode`s.
+-   Content outside of the tree, such as its schema objects and context, will not be reachable.
+-   No symbols will be encountered as keys. This ensures that the traversal can use the APIs which only support strings (like `Object.entries`) and get the same result as if using APIs which also support symbols (like `object spread`).
+
+Note that if using `for...in`, be sure to filter out `inherited` properties.
```

Comment:
Is this a requirement for our users? Or something we plan on offering tooling around?

Reply:
In practice, this is a requirement for anything authoring JavaScript or typescript code and its common to have a linter rule for it (we do), but its particularly relevant to this API as its designed to be compatible with the property enumeration approach listed in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties#traversing_object_properties  and of the 8 of them, `for...in` is the only case we don't just work as desired for due to including of inherited properties.

Reply:
Might be worth adding a brief code example of how to do the filtering (or link to one), then. I don't know about others, but I was not aware of this.

Reply:
if you open a ts file and type "forin" then hit tab vscode will give you an example that does it, but I have updated the doc to specifically recommend other approach (rather than explain how to do for-in properly, which is a bad idea)